### PR TITLE
feat: Support inline hightlight

### LINKED ISSUE #1343: Syntax highlight on inline code
### Is your feature request related to a problem? Please describe.

Support syntax highlight on inline code of docs. It would be cool when writing the type, default value ... of option

### Describe the solution you'd like

```md
## appearance
  - Type: `boolean`{ts}
  - Default: `true`{ts}
```
image

### Describe alternatives you've considered

Currently I can use the `transformHtml` hook to do it. But i think if vitepress could have built in support, it would be nice.

### Additional context

https://twitter.com/Atinux/status/1531266838583296001

image

https://github.com/nuxt/content/blob/b4e9591fc61c91551fb6e032f3bf62f8fe8e6159/docs/editor/useShiki.ts#L92-L108

### Validations

- [X] Follow our [Code of Conduct](https://vuejs.org/about/coc.html)
- [X] Read the [docs](https://vitepress.vuejs.org).
- [X] Read the [Contributing Guidelines](https://github.com/vuejs/vitepress/blob/main/.github/contributing.md).
- [X] Check that there isn't already an issue that asks for the same feature to avoid creating a duplicate.

### PR DESCRIPTION
**Pull request by VoVAllen**
Signed-off-by: Jinjing.Zhou <allenzhou@tensorchord.ai>

However, the color doesn't seem perfect in light mode. Dark mode works well

Fix #1343

### CODE REVIEW

File: docs/.vitepress/config.ts [hunk 1]
```diff
@@ -86,6 +86,7 @@ function sidebarGuide() {
       text: 'Writing',
       collapsible: true,
       items: [
+        { text: 'Inline highlight', link: '/guide/inline' },
```
Comment (Zhengqbbb) [inline docs/.vitepress/config.ts:89]: I don't think it is necessary to add an md file to explain
Can add headings in `docs/guide/markdown.md` in `Import Code Snippets` and `Markdown File Inclusion`

Comment (VoVAllen) [reply docs/.vitepress/config.ts:89]: sure. it's just to show the function for review.

Comment (Zhengqbbb) [reply docs/.vitepress/config.ts:89]: Well. Space split(or replace) possible is something wrong

```
- `const a: string = 'hello world'`{ts}
```

image

Comment (VoVAllen) [reply docs/.vitepress/config.ts:89]: Fixed. PTAL https://deploy-preview-1366--vitepress-docs.netlify.app/guide/inline

**Pull request closed without merging**